docs/api/index.rst

```diff
@@ -0,0 +1,11 @@
+Using the Hypothesis API
+########################
```

Comment:
A bit of a pedantic point here, but is there a reason this heading marker uses `#` while the other one uses `=`? I know rST doesn’t care which symbol you use, and that we aren’t consistent across the docs, but I think we should pick one and stick with it for future documentation changes, unless there’s a distinction I‘ve missed.